cli/cmd/createCluster.go

```diff
@@ -21,6 +27,219 @@ ksctl create-cluster ["azure", "gcp", "aws", "local"]
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
+var createClusterHACivo = &cobra.Command{
+	Use:   "ha-civo",
+	Short: "Use to create a HA CIVO k3s cluster",
+	Long: `It is used to create cluster with the given name from user. For example:
+
+ksctl create-cluster ha-civo <arguments to civo cloud provider>
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		isSet := cmd.Flags().Lookup("verbose").Changed
+		logger := log.Logger{Verbose: true}
+		if !isSet {
+			logger.Verbose = false
+		}
+		payload := civo.CivoProvider{
+			ClusterName: clusterName,
+			Region:      region,
+			HACluster:   true,
+			Spec: utils.Machine{
+				Disk:                nodeSize,
+				HAControlPlaneNodes: noControlPlaneNodes,
+				HAWorkerNodes:       noWorkerNodes,
+			},
+		}
+		err := payload.CreateCluster(logger)
+		if err != nil {
+			logger.Err(err.Error())
+			return
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
+var createClusterHAAzure = &cobra.Command{
+	Use:   "ha-azure",
+	Short: "Use to create a HA k3s cluster in Azure",
+	Long: `It is used to create cluster with the given name from user. For example:
+
+	ksctl create-cluster ha-azure <arguments to civo cloud provider>
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		isSet := cmd.Flags().Lookup("verbose").Changed
+		logger := log.Logger{Verbose: true}
+		if !isSet {
+			logger.Verbose = false
+		}
+
+		payload := &azure.AzureProvider{
+			ClusterName: clusterName,
+			HACluster:   true,
+			Region:      region,
+			Spec: util.Machine{
+				Disk:                nodeSize,
+				HAControlPlaneNodes: noControlPlaneNodes,
+				HAWorkerNodes:       noWorkerNodes,
+			},
+		}
+		err := payload.CreateCluster(logger)
+		if err != nil {
+			logger.Err(err.Error())
+			return
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
 func init() {
 	rootCmd.AddCommand(createClusterCmd)
+
+	createClusterCmd.AddCommand(createClusterAws)
+	createClusterCmd.AddCommand(createClusterAzure)
+	createClusterCmd.AddCommand(createClusterCivo)
+	createClusterCmd.AddCommand(createClusterLocal)
+	createClusterCmd.AddCommand(createClusterHACivo)
+	createClusterCmd.AddCommand(createClusterHAAzure)
+
+	createClusterAws.Flags().BoolP("verbose", "v", true, "for verbose output")
+	createClusterAzure.Flags().BoolP("verbose", "v", true, "for verbose output")
+	createClusterCivo.Flags().BoolP("verbose", "v", true, "for verbose output")
+	createClusterLocal.Flags().BoolP("verbose", "v", true, "Verbose output")
+	createClusterHACivo.Flags().BoolP("verbose", "v", true, "for verbose output")
+	createClusterHAAzure.Flags().BoolP("verbose", "v", true, "for verbose output")
+
+	createClusterAzure.MarkFlagRequired("name")
+	createClusterCivo.MarkFlagRequired("name")
+	createClusterCivo.MarkFlagRequired("region")
+	createClusterLocal.MarkFlagRequired("name")
+	createClusterHAAzure.MarkFlagRequired("name")
+	createClusterHACivo.MarkFlagRequired("name")
+
+	createClusterAzure.Flags().StringVarP(&clusterName, "name", "n", "", "Cluster name")
```

Comment:
need to find a better way to represent this, this looks long and messy. something like a whole seperate func to get/set flags

Reply:
added!